models/note.go

```diff
@@ -0,0 +1,34 @@
+// Package models defines the data types, payloads and repository interfaces
+// of the accounts service.
+package models
+
+import (
+	"context"
+
+	"github.com/google/uuid"
+)
+
+type NoteWithBlocks struct {
+	ID       uuid.UUID `json:"id" bson:"_id,omitempty"`
+	AuthorId string    `json:"authorId" bson:"authorId,omitempty"`
+	Title    string    `json:"title" bson:"title,omitempty"`
+	Blocks   []Block   `json:"blocks" bson:"blocks,omitempty"`
+}
+
+type NoteFilter struct {
+	ID       uuid.UUID `json:"id" bson:"_id,omitempty"`
+	AuthorId string    `json:"authorId" bson:"authorId,omitempty"`
+}
+
+// NotesRepository is safe for use in multiple goroutines.
+type NotesRepository interface {
+	Create(ctx context.Context, noteRequest *NoteWithBlocks) (*NoteWithBlocks, error)
+
+	Get(ctx context.Context, filter *NoteFilter) (*NoteWithBlocks, error)
+
+	Delete(ctx context.Context, filter *NoteFilter) error
+
+	Update(ctx context.Context, filter *NoteFilter, noteRequest *NoteWithBlocks) error
```

Comment:
A call to `Update` should return the updated representation of the object according to REST principles.
```suggestion
	Update(ctx context.Context, filter *NoteFilter, note *NoteWithBlocks) (*NoteWithBlocks, error)
```